kmm/kmm-post/src/commonMain/kotlin/com/gchristov/newsfeed/kmmpost/PostViewModel.kt

```diff
@@ -3,12 +3,18 @@ package com.gchristov.newsfeed.kmmpost
 import com.gchristov.newsfeed.kmmcommonmvvm.CommonViewModel
 import com.gchristov.newsfeed.kmmpostdata.PostRepository
 import com.gchristov.newsfeed.kmmpostdata.model.DecoratedPost
+import com.gchristov.newsfeed.kmmpostdata.usecase.DecoratePostUseCase
+import com.gchristov.newsfeed.kmmpostdata.usecase.ReadingTimeCalculationUseCase
 import kotlinx.coroutines.CoroutineDispatcher
 
 class PostViewModel(
     dispatcher: CoroutineDispatcher,
     private var postId: String,
+
+    //TODO: passing the repository into the viewmodel seems jumping ahead, it should
+    // be always repository -> use case -> viewmodel?
```

Comment:
This might be something for another PR if desired - In the Clean Architecture original spec normally all of the business logic is encapsulated in domain + use case as orchestrator, the repository stays normally within the usecase.

Reply:
It's a good point. However, the usage of the repository is quite light here - we're basically using its methods directly without applying any domain-specific logic other than mapping, so in this case there isn't much benefit of extracting this into a use case.

For decoration is different, as we're already starting to fetch data from multiple sources in order to decorate so it makes sense to split it out.

It's a bit of a balance 🙂